common/src/seg_tree.h

```diff
@@ -47,8 +85,26 @@ unsigned long seg_tree_max(struct seg_tree* seg_tree);
  *
  *    seg_tree_unlock(&seg_tree);
  */
+
+/*
+ * Lock a seg_tree for reading.  This should only be used for calling
+ * seg_tree_iter().  All the other seg_tree functions provide their
+ * own locking.
+ */
```

Comment:
This comment isn't true anymore since `seg_tree_find_nolock()` also requires the user to do locking.  Can you update all these comments?